sympy/solvers/solvers.py

```diff
@@ -849,13 +849,24 @@ def solve(f, *symbols, **flags):
     def _sympified_list(w):
         return list(map(sympify, w if iterable(w) else [w]))
     bare_f = not iterable(f)
+
+    # check flag usage
+    if flags.get('quick', None) is not None:
+        if flags.get('particular') == False:
+            raise ValueError('conflicting flags: particular=False and quick=True')
+        flags['particular'] = True
+    if flags.get('particular', False) and bare_f:
+        raise ValueError(filldedent("""
+            The 'particular/quick' flag is usually used with systems of
+            equations. Either pass your equation as a list or
+            consider using a solver like `diophantine` if you are
+            looking for a solution in integers."""))
```

Comment:
The flow control in this function is very complicated. I worry that this makes it worse.

Reply:
`quick` and `particular` are for the same solution path.
The options are particular/quick one of [True/True, True/False].
So if `quick` is set (i.e., not None) it sets `particular`.
Nothing ambiguous here.

If the user passes a single equation with `particular=True`, they are told that this is an error and redirected to what they probably are intending or to resend equation in a list.
Less ambiguity here.